### DISCUSSION TITLE
How to populate select using another select on query?

### DISCUSSION THREAD

**Discussion by pietro-giordano** [filamentphp/filament#10398] [Help] [answered]
Hi, i have this code and i want populate amount field when select product_brand field. Where do I go wrong?
I use filament v.2.
Ps. sorry for my english.

              Select::make('product_brand')
                    ->label('Brand')
                    ->options(Servizi::all()->where('productName', '=', 'Coupon')->pluck('productBrand')->unique())
                    ->searchable()
                    ->reactive()
                    ->afterStateUpdated(function (Closure $set, $state) {
                        $set('amount', Servizi::all()->where('productName', '=', 'Coupon')->where('productBrand', '=', $state)->pluck('amount'));
                    })
                    ->required(),

              Select::make('amount')
                  ->label('Costo')
                  ->required(),

**Comment by leandrocfe** [comment]
You should add the query in the options

```php
          Select::make('amount')
              ->label('Costo')
              ->required()
              ->options(fn($get):array => Servizi::whereProductBrand($get('product_brand') ?? null)->pluck('amount', 'id'))
```

**Reply by pietro-giordano** [reply]
Thanks, but it doesn't work :(

**Reply by pietro-giordano** [reply]
Unknown column 'product_brand' in 'where clause' (Connection: mysql, SQL: select `amount`, `id` from `servizi` where `product_brand` is null)

**Reply by pietro-giordano** [reply]
I have modified the code but the second select don't populate itself.

                Select::make('product_brand')
                    ->label('Brand')
                    ->options(Servizi::all()->where('productName', '=', 'Coupon')->pluck('productBrand')->unique())
                    ->searchable()
                    ->reactive()
                    ->afterStateUpdated(function (Closure $set, $state) {
                        $set('amount', Servizi::all()->where('productName', '=', 'Coupon')->where('productBrand', '=', $state)->pluck('amount'));
                    })
                    ->required(),

                Select::make('amount')
                    ->label('Costo')
                    ->options(fn ($get): array => Servizi::where('productBrand', $get('product_brand') ?? null)->pluck('amount', 'id')->toArray())
                    ->required(),

**Reply by leandrocfe** [reply, marked as answer]
```php
Select::make('product_brand')
    ->label('Brand')
    ->options(Servizi::where('productName','Coupon')->pluck('productBrand', 'productBrand')->unique())
    ->searchable()
    ->reactive()
    ->required(),

Select::make('amount')
    ->label('Costo')
    ->options(fn ($get): array => Servizi::where('productBrand', $get('product_brand') ?? null)->pluck('amount', 'id'))
    ->required(),
```
?

**Reply by pietro-giordano** [reply]
Yes, it works!! Thank you!!

               Select::make('product_brand')
                    ->label('Brand')
                    ->options(Servizi::all()->where('productName', '=', 'Coupon')->pluck('productBrand', 'productBrand')->unique())
                    ->searchable()
                    ->reactive()
                    ->required(),

                Select::make('amount')
                    ->label('Costo')
                    ->options(fn ($get): array => Servizi::where('productBrand', $get('product_brand') ?? null)->pluck('amount', 'id')->toArray())
                    ->required(),

The only think that I've added is toArray() method, because it doesn't work otherwise.
